packages/blocks/src/api/utils.js

```diff
@@ -121,3 +123,87 @@ export function normalizeBlockType( blockTypeOrName ) {
 
 	return blockTypeOrName;
 }
+
+/**
+ * Get the label for the block, usually this is either the block title,
+ * or the value of the block's `label` function when that's specified.
+ *
+ * @param {Object} blockType  The block type.
+ * @param {Object} attributes The values of the block's attributes.
+ * @param {Object} context    The intended use for the label.
+ *
+ * @return {string} The block label.
+ */
+export function getBlockLabel( blockType, attributes, context = 'visual' ) {
+	const {
+		__experimentalLabel: getLabel,
+		title,
+	} = blockType;
+
+	const label = getLabel && getLabel( attributes, { context } );
+
+	if ( ! label ) {
+		return title;
+	}
+
+	// Strip any HTML (i.e. RichText formatting) before returning.
+	return stripHTML( label );
+}
+
+/**
+ * Get a label for the block for use by screenreaders, this is more descriptive
+ * than the visual label and includes the block title and the value of the
+ * `getLabel` function if it's specified.
+ *
+ * @param {Object}  blockType  The block type.
+ * @param {Object}  attributes The values of the block's attributes.
+ * @param {?number} row        The row of the block in the block list.
+ *
+ * @return {string} The block label.
+ */
+export function getAccessibleBlockLabel( blockType, attributes, row ) {
+	// `title` is already localized, `label` is a user-supplied value.
+	const { title } = blockType;
+	const label = getBlockLabel( blockType, attributes, 'accessibility' );
+	const hasRow = row !== undefined;
+
+	// getBlockLabel returns the block title as a fallback when there's no label,
+	// if it did return the title, this function needs to avoid adding the
+	// title twice within the accessible label. Use this `hasLabel` boolean to
+	// handle that.
+	const hasLabel = label && label !== title;
+
+	if ( hasRow ) {
+		if ( hasLabel ) {
+			return sprintf(
+				/* translators: accessibility text. %s: block title, %d block row number, $s block label.. */
+				__( '%s Block. Row %d. %s' ),
+				title,
+				row,
+				label
+			);
+		}
+
+		return sprintf(
+			/* translators: accessibility text. %s: block title, %d block row number. */
+			__( '%s Block. Row %d' ),
```

Comment:
I used 'Row' in the text as that's what mobile called it previously, but it'd be great to make this consistent with the Block Mover accessibility labels, which call it 'Position'

Reply:
With horizontal movers, does it still make sense to use "Position" or would we be better with "Rows" and "Columns"?

Reply:
I think as soon as we enter the realm of 2d positioning, "position" is no longer sufficient, and rows and columns should be used.

Reply:
I've added this, but it won't work with every block. Many of the blocks that are horizontal are styled to display that way (like the columns block), there's no way to know from JS whether they actually are horizontal.

at the moment this should work in the navigation block where the blocks are laid out horizontally.